solutions/cses-1623.mdx

```diff
@@ -0,0 +1,150 @@
+
+        completeSearch(weights, 0, N, 0);
+	System.out.println((int)(result * 2));
+    }
+
+    public static void completeSearch(int[] weights, int i, int N,
+    	   long currentSum) {
+        if (i == N) {
+	    result = Math.min(result, Math.abs((total - currentSum)
+	    	   - currentSum) / 2.0);
+        } else {
+	    // option 1: not adding current index
+            completeSearch(weights, i + 1, N, currentSum);
+
+	    // option 2: adding current index
+            currentSubset.add(i);
+	    currentSum += weights[i];
+            completeSearch(weights, i + 1, N, currentSum);
+	    // undo option 2
+	    currentSum -= currentSubset.get(currentSubset.size() - 1);
+            currentSubset.remove(currentSubset.size() - 1);
+        }
+    }
+    CodeSnip{Kattio}
+}
+```
+</JavaSection>
+
+<CPPSection>
+```cpp
+#include<bits/stdc++.h>
+using namespace std;
+
+vector<int> currentSubset;
+// initialize result to max value because we want to get the minimum result
+double result = INT_MAX;
+long int total = 0;
+
+void completeSearch(vector<int> &weights, int i, int N, long int currentSum) {
+    if (i == N) {
+        result = min(result, abs((total - currentSum) - currentSum) / 2.0);
+    } else {
+        // option 1: not adding current index
+        completeSearch(weights, i + 1, N, currentSum);
+
+	// option 2: adding current index
+        currentSubset.push_back(i);
+        currentSum += weights[i];
+        completeSearch(weights, i + 1, N, currentSum);
+	// undo option 2
+        currentSum -= currentSubset[currentSubset.size() - 1];
+        currentSubset.pop_back();
+    }
+}
+
+int main() {
+    int N; cin >> N;
+    vector<int> weights(N);
+
+    for (int i = 0; i < N; i++) {
+        cin >> weights[i];
+	total += weights[i]; // calculate the total weight of all apples
+    }
+
+    completeSearch(weights, 0, N, 0);
+    cout << (int)(result * 2) << '\n';
+}
+```
+</CPPSection>
+
+<PySection>
+```py
+currentSubset = []
+# initialize result to max value because we want to get the minimum result
+result = 1e9
+total = 0
+
+def completeSearch(weights, i, N, currentSum):
+  global currentSubset
+  global result
+  global total
+  if i == N:
+    result = min(result, abs((total - currentSum) - currentSum) / 2.0);
+    # update the final result with absolute value of the difference between
+    # half of the total and the current sum
+  else:
+    # option 1: not adding current index
+    completeSearch(weights, i + 1, N, currentSum);
+    # option 2: adding current index
+    currentSubset.append(i);
+    currentSum += weights[i];
+    completeSearch(weights, i + 1, N, currentSum);
+    # undo option 2
+    currentSum -= currentSubset[len(currentSubset) - 1];
+    currentSubset.pop();
+
+N = int(input())
+weights = []
+
+for i in input().split(' '):
+    weights.append(int(i))
+    total += weights[len(weights)-1] # calculate the total weight of all apples
+
+completeSearch(weights, 0, N, 0)
+print((int)(result * 2))
```

Comment:
```suggestion
print(result * 2)
```
also uh just noticed why result * 2?

Reply:
oh i see you're... dividing it by 2 when calculating the result? is there a need for that?